src/main/java/de/craften/ui/swingmaterial/MaterialWindow.java

```diff
@@ -8,45 +8,439 @@
 import java.awt.geom.RoundRectangle2D;
 
 /**
- * A window with Material shadow.
+ * A {@code MaterialWindow} is the prefered top-level container for your
+ * Material applications. It includes its own nice Material shadow out of the
+ * box, as well as some convenient methods.
  */
 public class MaterialWindow extends JFrame {
+    /**
+     * This is the panel that contains the components placed inside this window.
+     * If you are wondering what happened to the JFrame's {@code contentPane},
+     * see {@link #shadowPane}.
+     */
+    private Container contentPane;
+    /**
+     * This panel is used as background of the window itself, so it drops a
+     * shadow.
+     * @see ShadowPane
+     */
+    protected ShadowPane shadowPane;
+    
+    /**
+     * Default constructor for a {@link MaterialWindow}. A container for the
+     * shadow is deployed if and only if the current graphics evironment
+     * supports per-prixel translucency for top-level containers (windows in
+     * general).
+     */
     public MaterialWindow() {
-        GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
-        if (ge.getDefaultScreenDevice().isWindowTranslucencySupported(GraphicsDevice.WindowTranslucency.PERPIXEL_TRANSLUCENT)) {
-            setUndecorated(true);
-            setBackground(new Color(255, 255, 255, 0));
-            setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-            setContentPane(new ShadowPane());
-            getRootPane().putClientProperty("Window.shadow", Boolean.FALSE);
-            getRootPane().setWindowDecorationStyle(JRootPane.NONE);
-            addComponentListener(new ComponentAdapter() {
-                @Override
-                public void componentResized(ComponentEvent componentEvent) {
-                    setShape(new RoundRectangle2D.Float(0, 0, getWidth(), getHeight(), 3, 3));
-                }
-            });
-        }
-    }
-
-    private class ShadowPane extends JPanel {
+        setUndecorated(true);
+        if (Utils.isTranslucencySupported()) {
+            setBackground(MaterialColor.TRANSPARENT);
+            //This default behavior should NOT be set here!
```

Comment:
Alright then.

We still need to figure out what keeps the VM running after disposing of all windows though.